tests/varispeed_test.c

```diff
@@ -144,8 +152,101 @@ varispeed_test (int converter, double target_snr)
 		exit (1) ;
 		} ;
 
-	puts ("ok") ;
-
 	return ;
 } /* varispeed_test */
 
+static void
+varispeed_hammer_test (int converter)
+{	double ratios [] = { 1.0, 0.01, 20 } ;
+	int chan, r1, r2 ;
+
+	for (chan = 1 ; chan <= 9 ; chan ++)
+		for (r1 = 0 ; r1 < ARRAY_LEN (ratios) ; r1++)
+			for (r2 = 0 ; r1 != r2 && r2 < ARRAY_LEN (ratios) ; r2 ++)
```

Comment:
Condition in the loop header does abort the loop instead of continuing it which results in not testing all combinations as intended